DBA_GA_utils.py

```diff
@@ -0,0 +1,278 @@
+            #print('Atom idx and N:',atom.GetAtomicNum(),atom.GetIdx())
+            for neigh in atom.GetNeighbors():
+                if neigh.GetAtomicNum() == 6:
+                    if neigh.GetIsAromatic() == True and neigh.IsInRing() == True:
+                        i += 1
+                        #print('Value of i =',i)
+        if atom.GetAtomicNum() == 0 and atom.GetIsotope() == 99:
+            dummies.append(atom.GetIdx())
+            #print('Atom idx and N:',atom.GetAtomicNum(),atom.GetIdx())
+            for neigh in atom.GetNeighbors():
+                if neigh.GetAtomicNum() == 6:
+                    if neigh.GetIsAromatic() == True and neigh.IsInRing() == True:
+                        j += 1
+                        #print('Value of j =',j)
+
+    #print('Dummies',len(dummies))
+    if i == 1 and j == 1 and mol.GetNumAtoms() > 18 and len(dummies) == 2:
+        return True
+    else:
+        return False
+
+def tweezer_classification(mol,side):
+    """
+    Function to classify how the boronic groups are bonded to the linker. The types 0, 1, and 2 means that 
+    the boronic groups are part of a PBA molecule in positions ortho, meta, or para, respectively. If the boronic
+    group is directly bonded to the backbone of the linker, this side is classified as type 3.
+    Input
+    mol: Rdkit molecule with the dummy atoms as *98 and *99
+    side: Strings "Glc" or "Frc" to indicate which side of the tweezer classify
+    Output
+    idx: 0, 1, 2, or 3
+    : tuple list of atoms with the coincidences for the past
+    """
+
+    oPBA_glcyl_SMARTS = ['[#6!$([CX3]=[OX1])!$([C][CX3]=[OX1])!$([C][NX3]),#7!$([NX3;H2,O2])!$([N+]([O-])[O-])!$([#7][CX3]=[OX1]),#8!$([O][CX4;H3]),#16!H1!$([#16X4](=[OX1])(=[OX1]))]~[#6R1,#7R1]1~[#6R1](-[98*])~[#6R1,#7R1]~[#6R1,#7R1]~[#6R1,#7R1]~[#6R1,#7R1]1', \
+                         '[#6!$([CX3]=[OX1])!$([C][CX3]=[OX1])!$([C][NX3]),#7!$([NX3;H2,O2])!$([N+]([O-])[O-])!$([#7][CX3]=[OX1]),#8!$([O][CX4;H3]),#16!H1!$([#16X4](=[OX1])(=[OX1]))]~[#6R1,#7R1]1~[#6R1,#7R1]~[#6R1,#7R1]~[#6R1,#7R1]~[#6R1,#7R1]~[#6R1]1(-[98*])']
+    mPBA_glcyl_SMARTS = ['[#6!$([CX3]=[OX1])!$([C][CX3]=[OX1])!$([C][NX3]),#7!$([NX3;H2,O2])!$([N+]([O-])[O-])!$([#7][CX3]=[OX1]),#8!$([O][CX4;H3]),#16!H1!$([#16X4](=[OX1])(=[OX1]))]~[#6R1,#7R1]1~[#6R1,#7R1]~[#6R1](-[98*])~[#6R1,#7R1]~[#6R1,#7R1]~[#6R1,#7R1]1', \
+                         '[#6!$([CX3]=[OX1])!$([C][CX3]=[OX1])!$([C][NX3]),#7!$([NX3;H2,O2])!$([N+]([O-])[O-])!$([#7][CX3]=[OX1]),#8!$([O][CX4;H3]),#16!H1!$([#16X4](=[OX1])(=[OX1]))]~[#6R1,#7R1]1~[#6R1,#7R1]~[#6R1,#7R1]~[#6R1,#7R1]~[#6R1](-[98*])~[#6R1,#7R1]1']
+    pPBA_glcyl_SMARTS = ['[#6!$([CX3]=[OX1])!$([C][CX3]=[OX1])!$([C][NX3]),#7!$([NX3;H2,O2])!$([N+]([O-])[O-])!$([#7][CX3]=[OX1]),#8!$([O][CX4;H3]),#16!H1!$([#16X4](=[OX1])(=[OX1]))]~[#6R1,#7R1]1~[#6R1,#7R1]~[#6R1,#7R1]~[#6R1](-[98*])~[#6R1,#7R1]~[#6R1,#7R1]1']
+    fPBA_glcyl_SMARTS = ['[#6R3,#6R2,#7X3R2,#16R2]~[#6R2,#7X3R2,#16R2]~[#6R1](-[98*])', \
```

Comment:
Finish to add the SMARTS expressions and add text to all exceptions in the code to have a record about the reason why the code does not worked correctly